src/receivers/HTTPModuleFunctions.ts

```diff
@@ -0,0 +1,264 @@
+      res.end(body);
+    } else {
+      res.writeHead(200, { 'content-type': 'application/json' });
+      res.end(JSON.stringify(body));
+    }
+  }
+
+  // ------------------------------------------
+  // Error handlers for event processing
+  // ------------------------------------------
+
+  // The default dispathErrorHandler implementation:
+  // Developers can customize this behavior by passing dispatchErrorHandler to the constructor
+  // Note that it was not possible to make this function async due to the limitation of http module
+  public static defaultDispatchErrorHandler(args: ReceiverDispatchErrorHandlerArgs): void {
+    const { error, logger, request, response } = args;
+    if ('code' in error) {
+      if (error.code === ErrorCode.HTTPReceiverDeferredRequestError) {
+        logger.info(`Unhandled HTTP request (${request.method}) made to ${request.url}`);
+        response.writeHead(404);
+        response.end();
+        return;
+      }
+    }
+    logger.error(`An unexpected error occurred during a request (${request.method}) made to ${request.url}`);
+    logger.debug(`Error details: ${error}`);
+    response.writeHead(500);
+    response.end();
+  }
+
+  // The default processEventErrorHandler implementation:
+  // Developers can customize this behavior by passing processEventErrorHandler to the constructor
+  public static async defaultProcessEventErrorHandler(
+    args: ReceiverProcessEventErrorHandlerArgs,
+  ): Promise<boolean> {
+    const { error, response, logger, storedResponse } = args;
+    if ('code' in error) {
+    // CodedError has code: string
+      const errorCode = (error as CodedError).code;
+      if (errorCode === ErrorCode.AuthorizationError) {
+      // authorize function threw an exception, which means there is no valid installation data
+        response.writeHead(401);
+        response.end();
+        return true;
+      }
+    }
+    logger.error('An unhandled error occurred while Bolt processed an event');
+    logger.debug(`Error details: ${error}, storedResponse: ${storedResponse}`);
+    response.writeHead(500);
+    response.end();
+    return false;
+  }
+
+  // The default unhandledRequestHandler implementation:
+  // Developers can customize this behavior by passing unhandledRequestHandler to the constructor
+  // Note that this method cannot be an async function to align with the implementation using setTimeout
+  public static defaultUnhandledRequestHandler(args: ReceiverUnhandledRequestHandlerArgs): void {
+    const { logger } = args;
+    logger.error(
+      'An incoming event was not acknowledged within 3 seconds. ' +
+      'Ensure that the ack() argument is called in a listener.',
+    );
+  }
+}
+
+export interface RequestVerificationOptions {
+  enabled?: boolean;
+  signingSecret: string;
+  nowMilliseconds?: () => number;
+  logger?: Logger;
+}
+
+// which handles errors occurred while dispatching a rqeuest
+export interface ReceiverDispatchErrorHandlerArgs {
```

Comment:
Renamed this a bit (just removed `HTTP` prefix) but the previous type name was not exposed to external code.